wordle-web/scripts/game.ts

```diff
@@ -47,15 +50,60 @@ export class Game {
     }
   }
 
+  public updateGuessedLetters() {
+    for (const letter of this.guess.letters) {
+      // Find the index of the letter in the guessed letters array
+      const index = this.guessedLetters.findIndex(
+        (existingLetter) => existingLetter.char === letter.char
+      );
+      if (index !== -1) {
+        // Do not update the letter if it is already correct
+        if (this.guessedLetters[index].state !== LetterState.Correct) {
+          // Do not update the letter if it is wrong
+          if (letter.state !== LetterState.Wrong) {
+            this.guessedLetters[index] = letter;
+          }
+        }
+      } else {
+        // If letter does not already exist, add it to the array
+        this.guessedLetters.push(letter);
+      }
+    }
+  }
+
+  public validWords() {
```

Comment:
So I think you guys are really really close on this. We made a separate ts page for the wordlist because of the complexity of the algorithm. You guys are on the right track and again we used a lot of helper methods like updatePage and a watch method in our vue page to update throughout the game!